core/dropdowndiv.js

```diff
@@ -744,5 +744,11 @@ DropDownDiv.repositionForWindowResize = function() {
     DropDownDiv.hide();
   }
 };
-
 exports = DropDownDiv;
+
+exports.testOnly_setGetBoundsInfo = function(getBoundsInfoMock) {
+  goog.setTestOnly();
+  getBoundsInfo = getBoundsInfoMock;
+};
```

Comment:
Ought we not to provide a mechanism for the test harness to restore the original value of `getBoundsInfo` once the test is over?  Perhaps doing so if `getBoundsInfoMock` is `null`?

Or does each test completely reload Blockly, such that a change to `getBoundsInfo` will not persist past the end of that specific test?

Reply:
Ah, this is a good point. AFAICT the tests do not completely reload Blockly. We would need something to restore the old value. However, I'm not entirely sure the best way to do this. I went spelunking through code and couldn't find any instances where they do this. Any ideas?

Reply:
Monica mentioned a way to do this would be to set a module-local variable and inside `getBoundsInfo` do
```
if (mockBoundsInfo) {
   mockBoundsInfo();
   return;
}
```
I'm a bit hesitant to add test logic to our code like this, so if we can't find a better solution I plan on chalking this up to a fun experiment and going with the original plan to just export the small number of private methods that we need in order to make our tests work.

Reply:
I will leave this to your judgement, as I think you understand the tests much better than I do.